language-plutus-core/stdlib/Language/PlutusCore/StdLib/Data/List.hs

```diff
@@ -174,3 +177,32 @@ getBuiltinSum s = do
         $ [ TyInst () (Constant () (BuiltinName () AddInteger)) $ TyInt () s
           , Constant () $ BuiltinInt () s 0
           ]
+
+-- |  'sumNat' as a PLC term.
+getBuiltinNatSum :: Natural -> Quote (Term TyName Name ())
+getBuiltinNatSum s = do
+    foldList <- getBuiltinFoldList
+    let int = TyBuiltin () TyInteger
+    let add = TyInst () (Constant () (BuiltinName () AddInteger)) $ TyInt () s
+    RecursiveType _ nat1 <- holedToRecursive <$> getBuiltinNat
+    nti <- getBuiltinNatToInteger s
+    acc <- freshName () "acc"
+    n <- freshName () "n"
+    RecursiveType _ nat2 <- holedToRecursive <$> getBuiltinNat
+    return
+        . foldl' (Apply ()) (foldl' (TyInst ()) foldList [nat1, int])
+        $ [ LamAbs () acc int $
+            LamAbs () n nat2 $
+            Apply () (Apply () add (Var () acc)) (Apply() nti (Var () n))
+          , Constant () $ BuiltinInt () s 0
+          ]
+
+-- | Convert a Haskell list of 'Term's to a PLC @list@.
+getListToBuiltinList :: Type TyName () -> [Term TyName Name ()] -> Quote (Term TyName Name ())
```

Comment:
I prefer to keep the PLC stdlib as PLC as possible, i.e. do not have many Haskell types there unless absolutely necessary like with `fixN`.

Reply:
We can have a `StdLib.Meta` module or something.